pygridgen/tests/test_focus.py

```diff
@@ -156,3 +156,25 @@ def test_focuspoint_to_dict():
     fp2 = pygridgen.grid._FocusPoint(**fp.to_dict())
     assert fp.to_dict() == {'pos': 0.25, 'axis': 'x', 'factor': 4, 'extent': 0.5}
     assert fp.to_dict() == fp2.to_dict()
+
+
+def test_focus_to_from_spec():
+    fpx = pygridgen.grid._FocusPoint(0.25, 'x', 4, 0.5)
+    fpy = pygridgen.grid._FocusPoint(0.33, 'y', 0.1, 0.2)
+    f1 = pygridgen.grid.Focus(fpx, fpy)
+
+    # Check that the spec dictionary returns desired results
+    assert f1.to_spec() == [
+        {'pos': 0.25, 'axis': 'x', 'factor': 4, 'extent': 0.5},
+        {'pos': 0.33, 'axis': 'y', 'factor': 0.1, 'extent': 0.2}
+    ]
+
+    f2 = pygridgen.grid.Focus.from_spec(f1.to_spec())
+
+    # Check that the spec dictionary returns desired results of the orignal
+    # focus and the focus generated from .to_spec()
+    # First check the keys are in order and then check the values
+    for dict1, dict2 in zip(f1.to_spec(), f2.to_spec()):
```

Comment:
I would simplify this by testing the behavior of the recreated object instead of it's properties.

```python
_x, _y = numpy.mgrid[1:5, 1:6]
x = _x / 5
y = _y / 6 # has to be between 0 & 1

x1, y1 = f1(x, y)
x2, y2 = f2(x, y)
nptest.assert_array_almost_equal(x1, x2)
nptest.assert_array_almost_equal(y1,yx2)
```